### DISCUSSION TITLE
Can't seem to be able to download artifacts using offline browsers like wget or curl

### DISCUSSION THREAD

**Discussion by JJ** [community/community#23996] [New to GitHub]
<p>I’m trying to create an <em>easy</em> download link for an artifact I’ve created (just in case you’re interested, my CV). Last URL for it is <a href="https://github.com/JJ/cv/suites/2221833274/artifacts/46005062" rel="noopener nofollow ugc">this one</a>. This will probably work perfectly when you click on it, however I can’t make it work using offline browsers like <code>wget</code> or <code>curl</code>. I’ve tried to set referer, to give it an user-agent, to follow redirects spanning different hosts (because it’s actually redirected to a different URL). No go. I can probably imitate a browser using some other offline tool, but I’d like to know what’s exactly the catch, since none of the usual tricks seem to work here.</p>

**Comment by byrneh** [comment]
<p>Hi <a class="mention" href="/jj">@JJ</a><br>
The hyperlink you provided <a href="https://github.com/JJ/cv/suites/2221833274/artifacts/46005062" rel="noopener nofollow ugc">https://github.com/JJ/cv/suites/2221833274/artifacts/46005062</a> returns a 404 not found in browser</p>
<p>If you want to get a single file using https you can use the following format<br>
(example from your repository below)<br>
curl <a href="https://raw.githubusercontent.com/JJ/cv/master/download_cv_url.txt" rel="noopener nofollow ugc">https://raw.githubusercontent.com/JJ/cv/master/download_cv_url.txt</a></p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Artifacts are only available to logged in users and have a limited retention period, so if you want an easy and persistent download link to give to people that’s probably not the right tool.</p>
<p>Maybe uploading the CV as a release would work for you?</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/github/administering-a-repository/about-releases" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/github/administering-a-repository/about-releases" target="_blank" rel="noopener">About releases - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/github/administering-a-repository/about-releases" target="_blank" rel="noopener">//docs.github.com/en/github/administering-a-repository/about-releases</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Reply by neilyoung** [reply]
Where do you read this statement in the link quoted?

**Reply by Simran-B** [reply]
About artifacts in general: https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts

The page about downloading artifacts with a browser does not explicitly state that you need to be logged in, it only states that you need read access to the repository: https://docs.github.com/en/actions/managing-workflow-runs/downloading-workflow-artifacts

**Reply by neilyoung** [reply]
OK, but I would think that it should work if I would authenticate my curl request, but it doesn't. Even though I'm the owner of the (private) repo. I always just get 404. But just for the single artefact I uploaded to the release (using Ansible). I can download the full source code, well, but not just the single binary artefact there. No issue in browser.

**Reply by neilyoung** [reply]
> The link I posted is about releases, it doesn't say anything about workflow artifacts

I'm also talking about releases.

**Reply by neilyoung** [reply]
<img width="1489" alt="image" src="https://user-images.githubusercontent.com/731020/187426094-cfc5becc-11a1-412f-9224-b774d8090613.png">

Here is my problem (just for explanation, don't bother if you don't like): My release. It has the source code as tar and zip and one additional binary tar.gz. While I'm able to download and unpack it in a browser, I'm not by no means able to do so using curl or wget. I'm always catching 404.

**Reply by airtower-luna** [reply]
That should really be a separate discussion, it's off-topic here. The key thing to additionally include in a new thread (if you choose to make one) is how you do authentication with curl or wget.

**Reply by neilyoung** [reply]
OK, thanks

EDIT: And bearer/token authentication using the PAT btw. I can completely see all meta data of my repository, just can't download. 

But ok, fine with it.

**Comment by JJ** [comment]
<p>But I didn’t know the problem was that you needed to be logged… That clearly explains why it’s not available to offline browsers.<br>
I’d have to go for the releases then. I tried to avoid that because I need to create a special token, upload it to the secrets… I think I’ll just assume that whoever downloads that is logged it… Thanks a lot!</p>

**Comment by JJ** [comment]
<p>No, it does not for me at least if I do it from the browser (and apparently while logged to GitHub, as pointed out by <a class="mention" href="/airtower-luna">@airtower-luna</a> )</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/jj">@JJ</a> 404 not found - means does not exist or the user does not have permissions when using GitHub. fyi I tried in the browser whilst logged in.<br>
A file placed in a ‘public’ repository is retrievable using https ‘and without authentication’ using the raw format example above.<br>
You can equally have as a downloadable asset in a release as described above.</p>
<p>Good luck with your endeavours</p>

**Comment by JJ** [comment]
<p>That might mean that only the original user has access to artifacts… Not good. Thanks for the advice. I didn’t know that.</p>

**Comment by airtower-luna** [comment]
<p>According to the <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token">Permissions for the <code>GITHUB_TOKEN</code> </a> list the <code>GITHUB_TOKEN</code> provided by the workflow should be able to create releases (via the <a href="https://docs.github.com/en/rest/reference/permissions-required-for-github-apps#permission-on-contents"><code>contents</code> permission</a>), without the need for another token.</p>
